pontoon/base/tests/managers/test_entity.py

```diff
@@ -1080,6 +1080,113 @@ def test_mgr_entity_filter_combined(admin, resource_a, locale_a, user_a):
     )
 
 
+@pytest.mark.django_db
+def test_mgr_entity_option_combined(admin, resource_a, locale_a, user_a):
+    """
+    Tests search options.
+    """
+    entities = [
+        EntityFactory.create(
+            key="key %s" % i,
+            resource=resource_a,
+            string="TestEntity%s" % i,
+        )
+        for i in range(0, 2)
+    ]
+
+    TranslationFactory.create(
+        locale=locale_a,
+        entity=entities[0],
+        string="TestEntity0",
+        rejected=True,
+        user=user_a,
+    )
+    TranslationFactory.create(
+        locale=locale_a,
+        entity=entities[1],
+        approved=True,
+        user=user_a,
+    )
+
+    # Base case
+    assert list(
+        Entity.for_project_locale(
+            admin,
+            resource_a.project,
+            locale_a,
+            search="",
+            author=user_a.email,
+        )
+    ) == [entities[i] for i in range(0, 2)]
+
+    # Test search_match_case
+    assert (
+        list(
+            Entity.for_project_locale(
+                admin,
+                resource_a.project,
+                locale_a,
+                search="testentity",
+                search_match_case=True,
+                author=user_a.email,
+            )
+        )
+        == []
+    )
```

Comment:
For each search option, it would be nice to have two tests - one where the option is set to True, and one where it is not. I would also organize each search option into a separate test case (function).